BDArmory/FX/ExplosionFX.cs

```diff
@@ -271,53 +270,72 @@ private List<BlastHitEvent> ProcessingBlastSphere()
+                                    {
+                                        if (explosionEventsVesselsHit.ContainsKey(damagedVesselName))
+                                            ++explosionEventsVesselsHit[damagedVesselName];
+                                        else
+                                            explosionEventsVesselsHit[damagedVesselName] = 1;
+                                    }
                                 }
                             }
                         }
-                    }
-                    else
-                    {
-                        DestructibleBuilding building = hitCollidersEnu.Current.GetComponentInParent<DestructibleBuilding>();
-
-                        if (building != null)
+                        else
                         {
-                            if (!explosionEventsBuildingAdded.Contains(building))
+                            DestructibleBuilding building = hitCollidersEnu.Current.GetComponentInParent<DestructibleBuilding>();
+
+                            if (building != null)
                             {
-                                ProcessBuildingEvent(building, explosionEventsPreProcessing, explosionEventsBuildingAdded);
+                                if (!explosionEventsBuildingAdded.Contains(building))
+                                {
+                                    //ProcessBuildingEvent(building, explosionEventsPreProcessing, explosionEventsBuildingAdded);
+                                    Ray ray = new Ray(Position, building.transform.position - Position);
+                                    var distance = Vector3.Distance(building.transform.position, Position);
+                                    RaycastHit rayHit;
+                                    if (Physics.Raycast(ray, out rayHit, Range * 2, explosionLayerMask))
+                                    {
+                                        //DestructibleBuilding destructibleBuilding = rayHit.collider.gameObject.GetComponentUpwards<DestructibleBuilding>();
+                                        distance = Vector3.Distance(Position, rayHit.point);
+                                        //if (destructibleBuilding != null && destructibleBuilding.Equals(building) && building.IsIntact)
+                                        if (building.IsIntact)
+                                        {
+                                            explosionEventsPreProcessing.Add(new BuildingBlastHitEvent() { Distance = distance, Building = building, TimeToImpact = distance / ExplosionVelocity });
+                                            explosionEventsBuildingAdded.Add(building);
+                                        }
+                                    }
+                                }
                             }
                         }
                     }
+                    catch
+                    {
+                    }
```

Comment:
Add in something like
```csharp
Debug.LogError($"[BDArmory.ExplosionFX]: Exception in overlapSphereColliders processing: {e.Message}\n{e.StackTrace}");
```
here. We want to find and fix any exceptions, not ignore them.

Reply:
Done.